### PR TITLE
drivers/ethernet/eth_gecko: auto-negotiate after link up

### PR DESCRIPTION
**Pull request by PierreAronnax**
Move auto-negotiate sequence from driver initialization to link up event
Previously when booting without ethernet cable connected the
initialization would fail and never recover.
Now we can connect the ethernet cable any time and multiple times.

This also drastically reduces boot time to main.

Logging Link up and Link down events.
Logging speed and duplex from eth_gecko logger instead of eth_gecko_phy.

Signed-off-by: Luuk Bosma <l.bosma@interay.com>

### CODE REVIEW

File: drivers/ethernet/eth_gecko.c [hunk 1]
```diff
@@ -60,6 +60,29 @@ static void link_configure(ETH_TypeDef *eth, u32_t flags)
...
+	if (result < 0) {
+		LOG_ERR("ETH PHY auto-negotiate sequence failed");
+		return;
+	}
+
+	LOG_INF("Speed %s Mb",
+		link_status & ETH_NETWORKCFG_SPEED ? "100" : "10");
+	LOG_INF("%s duplex",
+		link_status & ETH_NETWORKCFG_FULLDUPLEX ? "Full" : "Half");
+
```
Comment (jukkar) [inline drivers/ethernet/eth_gecko.c]: It would make sense to add `ETHERNET_AUTO_NEGOTIATION_SET | ETHERNET_DUPLEX_SET` capabilities to `eth_gecko_get_capabilities()` if we support changing these. That can be sent as a separate PR.

**Review by jukkar — APPROVED** [review]
LGTM

File: drivers/ethernet/eth_gecko.c (hunk 1 above)
Comment (PierreAronnax) [reply drivers/ethernet/eth_gecko.c]: Agreed. Because this PR changes this capability I have included it in this PR.

**Comment by Oanerer** [comment]
Verified that it's working as intended! Great, thanks!
